src/array.cr

```diff
@@ -1808,6 +1811,21 @@ class Array(T)
     insert 0, obj
   end
 
+  # Prepend multiple values. The same as `unshift`, but takes an arbitrary number
+  # of values to add to the array. Returns `self`.
+  def unshift(*values : T)
+    new_size = @size + values.size
+    resize_to_capacity(Math.pw2ceil(new_size)) if @size > @capacity
+    move_value = values.size
+    @buffer.move_from(@buffer - move_value, @size + move_value)
```

Comment:
This is wrong, you can't move from `@buffer - move_value` that's unknown memory for the array. This should be:

```
(@buffer + move_value).move_from(@buffer, move_value)
```